x-pack/plugins/secops/public/components/timeline/body/renderers/auditd_executed_details.test.tsx

```diff
@@ -0,0 +1,135 @@
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import toJson from 'enzyme-to-json';
+import * as React from 'react';
+import { mountWithIntl, shallowWithIntl } from 'test_utils/enzyme_helpers';
+
+import { mockEcsData, TestProviders } from '../../../../mock';
+
+import { AuditdExecutedCommandLine, AuditdExecutedDetails } from './auditd_executed_details';
+
+describe('AuditExecutedDetails', () => {
+  describe('rendering', () => {
+    test('it renders the default AuditExecutedDetails', () => {
+      const wrapper = shallowWithIntl(
+        <TestProviders>
+          <AuditdExecutedDetails data={mockEcsData[19]} />
+        </TestProviders>
+      );
+      expect(toJson(wrapper)).toMatchSnapshot();
+    });
+
+    test('it returns auditd executed if the data does contain auditd executed data', () => {
+      const wrapper = mountWithIntl(
+        <TestProviders>
+          <AuditdExecutedDetails data={mockEcsData[19]} />
+        </TestProviders>
+      );
+      expect(wrapper.text()).toEqual('johnson@zeek-sanfran:/>gpgconf--list-dirs agent-socket');
+    });
+
+    test('it returns null for text if the data contains no auditd executed data', () => {
+      const wrapper = mountWithIntl(
+        <TestProviders>
+          <AuditdExecutedDetails data={mockEcsData[0]} />
+        </TestProviders>
+      );
+      expect(wrapper.text()).toEqual(null);
+    });
+  });
+
+  // NOTE: It's best if all the arguments are sent into this function and they typically should be otherwise
+  // you have something wrong with your beats. These tests are to ensure the function does not
+  // crash. If you need to format things prettier because not all the data is there, then update
+  // these tests with those changes
+  describe('#AuditdExecutedCommandLine', () => {
+    test('it returns pretty output if you send in all your happy path data', () => {
+      const wrapper = mountWithIntl(
+        <TestProviders>
+          <AuditdExecutedCommandLine
+            id={'hello-i-am-an-id'}
+            hostName={'host-1'}
+            userName={'username-1'}
+            processName={'process-1'}
+            processTitle={'process-title-1'}
+            workingDirectory={'working-directory-1'}
+            args={'arg1 arge 2 arg3'}
+          />
+        </TestProviders>
+      );
+      expect(wrapper.text()).toEqual(
+        'username-1@host-1:working-directory-1>process-1arg1 arge 2 arg3'
+      );
+    });
+
+    test('it returns ugly output if you just send in an id', () => {
+      const wrapper = mountWithIntl(
+        <TestProviders>
+          <AuditdExecutedCommandLine id={'hello-i-am-an-id'} />
+        </TestProviders>
+      );
+      expect(wrapper.text()).toEqual('@:>');
+    });
+
+    test('it returns slightly less ugly output if you just send in an id and hostName', () => {
```

Comment:
I've seen deprecation warnings in code before, but I've never seen _self-deprecating_ code, until these tests!

Reply:
You can ping me (technically) on it with a bug report that I should be showing `--` (emptyValue()) and assign it to me so if we see a problem we can D&D into the timeline showing the `--`. We haven't fixed timeline to work with `--` yet though and I don't know why this data would be missing (at this point) but as we have seen with port numbers, sometimes you just don't have that data from the logs.